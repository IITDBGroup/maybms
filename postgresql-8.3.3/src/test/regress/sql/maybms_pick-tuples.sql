--test of pick-tuples

create table r(a int, b real);
insert into r values (1,2),(1,3),(2,-1),(2,0);

--syntax check
pick tuples from r independently;

pick tuples from r independently with probability 1;

pick tuples from r with probability 1;

--raise an error for p > 1
pick tuples from r with probability b;

drop table r;
create table r(a int, b real);
insert into r values (1,0.5),(1,0.4),(2,-1),(2,0);

--raise an error for p < 0
pick tuples from r with probability b;

--silently remove tuples with probability 0
pick tuples from r with probability (b+1)/10;

drop table r;


