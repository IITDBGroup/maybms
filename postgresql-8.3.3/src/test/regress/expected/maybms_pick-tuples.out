--test of pick-tuples
create table r(a int, b real);
insert into r values (1,2),(1,3),(2,-1),(2,0);
--syntax check
pick tuples from r independently;
 a | b  | _v0 | _d0 | _p0 
---+----+-----+-----+-----
 1 |  2 |   1 |   1 | 0.5
 1 |  3 |   2 |   1 | 0.5
 2 | -1 |   3 |   1 | 0.5
 2 |  0 |   4 |   1 | 0.5
(4 rows)

pick tuples from r independently with probability 1;
 a | b  | _v0 | _d0 | _p0 
---+----+-----+-----+-----
 1 |  2 |   5 |   1 |   1
 1 |  3 |   6 |   1 |   1
 2 | -1 |   7 |   1 |   1
 2 |  0 |   8 |   1 |   1
(4 rows)

pick tuples from r with probability 1;
 a | b  | _v0 | _d0 | _p0 
---+----+-----+-----+-----
 1 |  2 |   9 |   1 |   1
 1 |  3 |  10 |   1 |   1
 2 | -1 |  11 |   1 |   1
 2 |  0 |  12 |   1 |   1
(4 rows)

--raise an error for p > 1
pick tuples from r with probability b;
ERROR:  Invalid probability:2.000000. The probability must be from (0,1]. 
drop table r;
create table r(a int, b real);
insert into r values (1,0.5),(1,0.4),(2,-1),(2,0);
--raise an error for p < 0
pick tuples from r with probability b;
ERROR:  Invalid probability:-1.000000. The probability must be from (0,1]. 
--silently remove tuples with probability 0
pick tuples from r with probability (b+1)/10;
 a |  b  | _v0 | _d0 | _p0  
---+-----+-----+-----+------
 1 | 0.5 |  15 |   1 | 0.15
 1 | 0.4 |  16 |   1 | 0.14
 2 |   0 |  17 |   1 |  0.1
(3 rows)

drop table r;
